src/types/shared-types.ts

```diff
@@ -0,0 +1,23 @@
+import { Field } from './formbuilder-types'
+
+//Need Kevin to review for completeness
+export interface LayoutTemplates {
+  label?: (label: string, data: any) => any
+  help?: (helpText: string) => any
+  default?: (field: Field, label: JQuery<HTMLElement>, help: JQuery<HTMLElement>, data: any) => any
+  noLabel?: (field: Field, label: JQuery<HTMLElement>, help: JQuery<HTMLElement>, data: any) => any
+  hidden?: (field: Field) => any
+
+  configure?: () => any
```

Comment:
i'm not sure about `configure` and `build`, it looks like they were meant to be extended differently. Possibly because they are not technically templates they should be extended separately but I think the method of extension should be the same. In other words we shouldn't do prototype extension _and_ option extension for different parts of the same class.

I'm generally not a fan of extending prototype when it can be avoided.